internal/ethapi/api.go

```diff
@@ -943,8 +949,36 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 	// this makes sure resources are cleaned up.
 	defer cancel()
 
+	enrichedCtx := ctx
+	// create callbacks to support runtime multitenancy checks during the run
+	if authToken, ok := b.SupportsMultitenancy(ctx); ok {
+		var authorizeMessageCallFunc multitenancy.AuthorizeMessageCallFunc = func(contractAddress common.Address) (bool, bool, error) {
```

Comment:
Could we not add the `AuthorizeMessageCallFunc` method on the `Backend` interface? For example a method named `IsAuthorizedMessage`.

Reply:
Callback implementation is different for `eth_call` and simulation. Also the data required for the closure are sourced in the same function.